### PR TITLE
Changes to Home Page Formatting

### PR DESCRIPTION
**Pull request by Pulathisi**
#### Description

Changing the home page formatting. 

####

- [ ] New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?

yarn test done. All tests passed

#### Checklist:

- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: src/main/webapp/app/home/home.component.scss [hunk 1]
```diff
@@ -30,56 +30,59 @@ Main page styles
   margin-top: 15%;
 }
 
-#mission {
-  background-color: white;
-  padding: 1rem 0 1rem 0;
-}
+// #mission {
```
Comment (SudharakaP) [inline src/main/webapp/app/home/home.component.scss]: You don't need to comment them out; just delete them if they are unnecessary. :smile:

**Comment by SudharakaP** [comment]
@Pulathisi : Can you please fill out the description of the PR with what it does and change the heading too into what the PR is about. :smile: 

Also I saw that you've added a `package-lock.json` file. This means you've used `npm`. However the project uses [`yarn`](https://yarnpkg.com/lang/en/). Could you remove that file and use `yarn` instead. Yarn works almost the same way as npm; https://github.com/python-sinhala-education-society/OpenLearnr/blob/master/docs/DEVELOPER_DOCUMENTATION.md

File: package.json [hunk 2]
```diff
@@ -36,6 +36,7 @@
     "stripe-angular": "0.5.0",
     "swagger-ui": "2.2.10",
     "tslib": "1.10.0",
+    "yarn": "^1.19.1",
```
Comment (SudharakaP) [inline package.json]: This is unnecessary. The generator of this project auto installs yarn locally. If you go to your root folder of the project you'll see a npm directory and within it a yarn directory. This is where yarn is installed. :smile:

File: src/main/webapp/content/css/overrides.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* navbar */
+
+.jh-navbar {
+  background-color: #222035 !important;
```
Comment (SudharakaP) [inline src/main/webapp/content/css/overrides.css]: Can we not use this !important annotation when making changes, but rather directly change [.jh-navbar](https://github.com/python-sinhala-education-society/OpenLearnr/blob/b3da28ad4b644f29cec863dc96d79cfa779742b0/src/main/webapp/app/layouts/navbar/navbar.scss#L8) and [.hipster](https://github.com/python-sinhala-education-society/OpenLearnr/blob/b3da28ad4b644f29cec863dc96d79cfa779742b0/src/main/webapp/app/home/home.scss#L5). It makes it difficult to maintain the code in the future; https://css-tricks.com/when-using-important-is-the-right-choice/

**Comment by SudharakaP** [comment]
@Pulathisi : I tested your change and it seems to remove some formatting from the front page. Also the mobile formatting seems off. 

image
  
whereas the original one was, 

image

Also the change seems to make the titles and text centered;

image

whereas the original was,

image

Normally card text and titles are left aligned (https://material.io/components/cards/#). I try to maintain the Google Material Design guide; so can we go by it for these kind of changes. :smile: 

Can we not do this change for now machn? I don't see a huge improvement to the front page. One way we could improve it though is making it more in accordance with [material design guidelines](https://material.io).

File: package.json (hunk 2 above)
Comment (Pulathisi) [reply package.json]: yarn suggested to do 'yarn audit autofix' to fix some yarn irregularities. I did that and yarn has added this.

File: src/main/webapp/content/css/overrides.css (hunk 3 above)
Comment (Pulathisi) [reply src/main/webapp/content/css/overrides.css]: changed css to navbar.scss and deleted overrides.css file ;) change come in the next sync

**Comment by Pulathisi** [comment]
> @Pulathisi : I tested your change and it seems to remove some formatting from the front page. Also the mobile formatting seems off.
> 
> image
> 
> whereas the original one was,
> 
> image
> 
> Also the change seems to make the titles and text centered;
> 
> image
> 
> whereas the original was,
> 
> image
> 
> Normally card text and titles are left aligned (https://material.io/components/cards/#). I try to maintain the Google Material Design guide; so can we go by it for these kind of changes. 😄
> 
> Can we not do this change for now machn? I don't see a huge improvement to the front page. One way we could improve it though is making it more in accordance with [material design guidelines](https://material.io).

@SudharakaP LOL is this material. I thought its flat design. Let me know if you want a material touch to it. U have to add shadows and stuff

https://en.wikipedia.org/wiki/Flat_design

**Comment by Pulathisi** [comment]
@SudharakaP I force pushed my changes related to home page. Let me know if there is any issues with that ;) check for mobile devices now. I did some ui modifications. Also please note that I didnt check for Ipads. Only mobiles

**Comment by SudharakaP** [comment]
@Pulathisi : Hey, machn, sorry; can we not do this change for now? The reason is our front page and stuff is okay for now and it works without issue; surely it can be improved but there's more pressing issues with the site than changing the look of the front page. I'll keep this in mind and we'll revisit this later if you are okay with that. :smile: 

It's my fault actually and I should have been more clear of the process; sorry if it wasted your time. Normally what I do is this; I go to issues (https://github.com/python-sinhala-education-society/OpenLearnr/issues) and create a new issue with the changes I want to do. Sometimes @asanka-herath will do that too. Also I normally add the appropriate badges and include it in our project board (https://github.com/python-sinhala-education-society/OpenLearnr/projects/2). And when I have time I go to this project board and take an issue and work on it. This is how normally most open source projects work. The issue is for discussing the change and then we do the pull request if it's agreed to do it. 

How about this? For front end stuff, there's lots of things to do actually. One of the things that @asanka-herath has problems with is searching for videos from the comment approval screen. We have an issue for that; #51 . Also #116 is a good improvement that could make things easier for admins. In addition the following two issues that I just created; #154, #155 could improve user experience for the admin/instructors greatly. If you have any questions about these feel free to comment on those issues. Also feel free to create any new issues if you have new ideas. :smile: I am sure there's lots of this kind of issues; and I'll create some more in the days to come. 

The aesthetic appeal of the website is okay for now. We lack lots of features such as notifications when a comment is made; and [comment editing #55](https://github.com/python-sinhala-education-society/OpenLearnr/issues/55) and after we do these we'll come back to polishing the UI aspects.

**Comment by Pulathisi** [comment]
@SudharakaP Its now I realized what you really intend me to do. 😄 I will work on the issues you are referring to but since you are saying the UI can be improved I doubt why you don't go ahead with these changes. For me the mobile UIs and Desktop UIs have improved with my changes and I don't see any reason to reject these changes. If front page stuff is okay but surely can be improved shall we accept these changes and move on from here ?

**Comment by SudharakaP** [comment]
> @SudharakaP Its now I realized what you really intend me to do. smile I will work on the issues you are referring to but since you are saying the UI can be improved I doubt why you don't go ahead with these changes. For me the mobile UIs and Desktop UIs have improved with my changes and I don't see any reason to reject these changes. If front page stuff is okay but surely can be improved shall we accept these changes and move on from here ?

The reason is trying to merge this we'll have to do few more iterations of testing and corrections. For example the [CI is failing with this change](https://travis-ci.org/python-sinhala-education-society/OpenLearnr/builds/600332300?utm_source=github_status&utm_medium=notification). I like us to work on things that being currently are problems for users (such as the ones I described in my previous comment). And I feel we have more pressing matters to address than doing visual changes right now and keep iterating on this issue wastes both our time. Does that make sense? :smile: :smile: 

However if you really want to go ahead with this change, let me know as well; we cannot merge this as it is will have to do some more corrections perhaps (such as getting the CI to pass). Also if you don't like the issues I mentioned above please feel free to just open an issue with your ideas then we could discuss about them. :smile:

**Comment by SudharakaP** [comment]
@Pulathisi : I am going to close this issue for now; I've added some more info to our [readme](https://github.com/python-sinhala-education-society/OpenLearnr/blob/develop/README.md) and [contributions pages](https://github.com/python-sinhala-education-society/OpenLearnr/blob/master/docs/CONTRIBUTING.md) about opening an issue before making pull requests. This way in the future we can all be in the same page. 😄

**Comment by Pulathisi** [comment]
@SudharakaP If that's the case I am not the one suitable for this project. 😄  Because front end look and feel is what I enjoy working with. So best of luck with your project. Wish you guys strength to continue the good work. 😄

**Comment by SudharakaP** [comment]
@Pulathisi : How about this. If you like to work on front end UI changes can you open an issue on how it can be improved, maybe even linking to this PR? Is this PR a larger change you anticipate to get the site in accordance with some sort of design you have in mind? :smile: 

The problem I am thinking is with the PR first approach it's harder to understand what the change achieves. What I am suggesting is exactly what we do for example in this project; https://github.com/jhipster/generator-jhipster.

**Pull request closed without merging**